api/dp-openapi.yaml

```diff
@@ -733,6 +845,63 @@ components:
       in: header
       name: X-API-KEY
 
+  parameters:
+    projectid:
+      name: projectid
+      in: path
+      description: ID of project
+      required: true
+      schema:
+        type: string
+
+    projectstate:
+      name: projstate
+      in: query
+      description: Project state
+      required: true
+      schema:
+        type: string
+
+    pagename:
+      name: pagename
+      in: path
+      description: name of page
+      required: true
+      schema:
+        type: string
+
+    pagestate:
+      name: pagestate
+      in: query
+      description: page state
+      required: true
+      schema:
+        type: string
+
+    pagefunction:
+      name: pagefunction
+      in: query
+      description: action to perform on the page.<br><br>
+        resume If the page is saved-as-done 'unsaves' it, gets data for the given page.<br><br>
+        saveasdone Checks the page for invalid characters and if good, saves it as done. Else throws an exception.<br><br>
+        savetemp Checks the page for invalid characters and if good, saves it as in progress, returns page text and state. Else throws an exception.<br><br>
+        saverevert Checks the page for invalid characters and if good, saves it as in progress, returns original page text and current state. Else throws an exception.<br><br>
+        return Returns the page to the round.
+      required: true
+      schema:
+        type: string
+
+  requestBodies:
+    TextBody:
```

Comment:
Consider `page_text` so it mirrors other `page_text_*` parameters.

Reply:
done this